dvc/objects/diff.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+if TYPE_CHECKING:
+    from .file import HashFile
+
+ADD = "add"
+MODIFY = "modify"
+DELETE = "delete"
+UNCHANGED = "unchanged"
+
+
+@dataclass
+class KeyObjPair:
+    in_cache: bool
+    key: Tuple[str]
+    obj: Optional["HashFile"] = field(default=None)
+
+    def __bool__(self):
+        return bool(self.obj)
+
+    def __eq__(self, other):
+        if not isinstance(other, KeyObjPair):
+            return False
+
+        if self.key != other.key or bool(self.obj) != bool(other.obj):
+            return False
+
+        if not self.obj:
+            return False
+
+        return self.obj.hash_info == other.obj.hash_info
+
+
+@dataclass
+class Change:
+    old: KeyObjPair
+    new: KeyObjPair
```

Comment:
What's `old` and what's `new` in the objects?

Reply:
Not sure I understand the question. Could you elaborate, please? `old` and `new` are `diff` terms, rather than `object` terms. E.g. dulwich uses `tree1` and `tree2` https://github.com/dulwich/dulwich/blob/81bd07a84254036af7e95268921d653450aa4b3e/dulwich/diff_tree.py#L166 , but I find that more confusing than old and new, as with the latter you at least instinctively know the direction of the difference.